src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -130,14 +138,39 @@
     <Field Name="IsNewInstance" Type="bool" Null="NotApplicable"/>
   </Node>
 
-  <!--
-  This node is used to represent an expression of a certain type, when attempting to bind its Index or Range pattern indexer
-  It does survive past initial binding but will be replaced with a synthesized argument in DAG lowering.
-  -->
+  <!-- This node is used to represent the argument of type Index or Range for an implicit Index or Range indexer. It does not survive lowering -->
   <Node Name="BoundIndexOrRangeIndexerPatternValuePlaceholder" Base="BoundValuePlaceholderBase">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
   </Node>
 
+  <!-- This node is used to represent the receiver for an implicit Index or Range indexer. It does not survive lowering -->
+  <Node Name="BoundIndexOrRangeIndexerPatternReceiverPlaceholder" Base="BoundValuePlaceholderBase">
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="ValEscape" Type="uint" Null="NotApplicable"/>
+  </Node>
+
+  <!-- This node represents the receiver for a list pattern. It does not survive lowering -->
```

Comment:
>It does not survive lowering [](http://example.com/codeflow?start=61&length=28)

Consider updating the comments on these, since they should be gone prior to lowering.

Reply:
I'll leave that as-is, as technically the placeholder replacer is run as part of local rewriting (which is part of lowering).
By deriving from BoundEarlyValuePlaceholderBase we're indicating that it's replaced early.